dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report.repo.test.ts

```diff
@@ -0,0 +1,55 @@
+import { when } from 'jest-when';
+import { Not } from 'typeorm';
+import { UtilisationReportEntityMockBuilder, UtilisationReportReconciliationStatus } from '@ukef/dtfs2-common';
+import { UtilisationReportRepo } from './utilisation-report.repo';
+
+describe('UtilisationReportRepo', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('findReportingYearsByBankId', () => {
+    const findSpy = jest.spyOn(UtilisationReportRepo, 'find');
+
+    beforeEach(() => {
+      findSpy.mockRejectedValue('Some error');
+    });
+
+    it('returns a set containing all the years where the report status is not REPORT_NOT_RECEIVED', async () => {
+      // Arrange
+      const bankId = '123';
+      const reports = [
+        UtilisationReportEntityMockBuilder.forStatus('RECONCILIATION_IN_PROGRESS')
+          .withBankId(bankId)
+          .withReportPeriod({
+            start: { month: 1, year: 2020 },
+            end: { month: 1, year: 2021 },
+          })
+          .build(),
+        UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION')
+          .withBankId(bankId)
+          .withReportPeriod({
+            start: { month: 1, year: 2023 },
+            end: { month: 1, year: 2023 },
+          })
+          .build(),
+      ];
+
+      when(findSpy)
+        .calledWith({
+          where: {
+            bankId,
+            status: Not<UtilisationReportReconciliationStatus>('REPORT_NOT_RECEIVED'),
+          },
+          select: ['reportPeriod'],
+        })
+        .mockResolvedValue(reports);
```

Comment:
That is true, but typescript would be annoyed at me doing that. However, I think because we are purposefully doing that here it does make more sense to only include that property and do some explicit generics (type-hacking, if you will) to make typescript happy